### PR TITLE
azurerm: Add AzureAD Auth

### LINKED ISSUE #143: Feature Request: Azure - Support AAD Auth
Please add support for `use_azuread_auth` (+ parameter on the URL)

https://developer.hashicorp.com/terraform/language/settings/backends/azurerm#use_azuread_auth

Here is an example to authenticate without using a Storage Account Key (https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/blob/examples_test.go#authenticate-the-client)

and `azidentity.NewDefaultAzureCredential(nil)` is already used.

https://github.com/fujiwara/tfstate-lookup/blob/b2cc81e5c97df6328e3f30a4fd0c9ea095b93d9e/tfstate/remote_azurerm.go#L102

### PR DESCRIPTION
**Pull request by jkroepke**
fixes #143

### CODE REVIEW

File: tfstate/remote_azurerm.go [hunk 1]
```diff
@@ -121,3 +133,24 @@ func getDefaultAccessKey(ctx context.Context, resourceGroupName string, accountN
...
+
+	subscriptionID, err := getDefaultSubscription()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get default subscription")
+	}
+
+	return subscriptionID, nil
+}
+
+func getDefaultClient() (*azidentity.DefaultAzureCredential, error) {
```
Comment (fujiwara) [inline tfstate/remote_azurerm.go:150]:
```suggestion
func getDefaultAzureCredential() (*azidentity.DefaultAzureCredential, error) {
```

This function does not return "Client", seems to return "Credential".
Would you consider renaming the function?

**Review by fujiwara — COMMENTED** [review]
@jkroepke Thank you!
Added a comment for naming functions.

And I noticed names of functions for Azure should include "azure". I'll refactor that with another PR.

File: tfstate/remote_azurerm.go (hunk 1 above)
Comment (jkroepke) [reply tfstate/remote_azurerm.go]: Done!

**Comment by fujiwara** [comment]
@jkroepke Thank you! I've confirmed works fine.

**Comment by fujiwara** [comment]
v1.1.5 Released! https://github.com/fujiwara/tfstate-lookup/releases/tag/v1.1.5

**Pull request merged**